service/timeline.go

```diff
@@ -113,16 +107,28 @@ func (uis *UIServer) patchTimelineJson(w http.ResponseWriter, r *http.Request) {
 			Project(patch.ExcludePatchDiff).
 			Sort([]string{"-" + patch.CreateTimeKey}).
 			Skip(skip).Limit(DefaultLimit))
+		if err != nil {
+			uis.LoggedError(w, r, http.StatusInternalServerError, errors.Wrapf(err,
+				"Error fetching patches for user %v", user))
+			return
+		}
 	} else {
+		project_id := gimlet.GetVars(r)["project_id"]
+		project, err := projCtx.GetProject()
+		if err != nil || project == nil {
+			uis.LoggedError(w, r, http.StatusInternalServerError, errors.Wrapf(err,
```

Comment:
maybe we'd want to return a 404 in the case where `project == nil`?

Reply:
Done!

Reply:
I'm sorry: I realized I made a mistake. I don't think `GetProject` will ever return a nil project if there's no error.

Reply:
but maybe it still makes sense to check if 
https://github.com/evergreen-ci/evergreen/blob/77c561e30a34a749135dd8cc5462f5730396ae01/model/project_ref.go#L177
returns nil?

Reply:
Combined the errors, but leaving a check for whether project is nil because we dereference it later (even if it seems like GetProject will always return a pointer).

Reply:
In this case, we could theoretically skip GetProject and use the projectID directly.

Reply:
Ah, I was too hasty to merge! I looked into it and think we should get the project identifier from the context, because the value of the project_id request variable could be overwritten by the default project or by a cookie value in this function:

https://github.com/evergreen-ci/evergreen/blob/77c561e30a34a749135dd8cc5462f5730396ae01/service/middleware.go#L285

I think we should use the project context as the ultimate source of truth?

Reply:
sounds good to me!